promscale/query/query-traces.md

```diff
@@ -0,0 +1,425 @@
+
+Such queries can be very useful first exploratory step into a new system. Typicall though the volume and diversity of trace data in modern systems is very high, at the very least we want to limit the scope of our query.
+
+## 3.2. Filtering <a name="para-3-2"></a>
+
+To filter the data we're interested in we're going to use the SQL `where` clause:
+
+```SQL
+select
+        trace_id,
+        span_id,
+        status_message
+    from span
+    where start_time >= now() - interval '30 minutes'
+```
+Here we limit our query to spans that happened within last 30 minutes. This type of query is far more efficient than those without any filters in the `where` clause, as it allows the optimizer do it's magic, namely eliminating unnecessary [chunks from the hypertable](https://docs.timescale.com/timescaledb/latest/overview/core-concepts/hypertables-and-chunks/#partitioning-in-hypertables-with-chunks=) and using indexes to locate rows that satisfy our quals.
+
+When dealing with timeseries data it is very important to define the time window that is of interest, because the data is partitioned based on the time and thus omitting time filters can drastically hinder the performance of the queries.
+
+When dealing with traces it can be extremely important to pinpoint a particular trace tag. This can be done by using standard PostgreSQL [`json` operators](https://www.postgresql.org/docs/current/functions-json.html#FUNCTIONS-JSONB-OP-TABLE):
+
+```SQL
+select *
+    from span
+    where 
+            start_time >= now() - interval '30 minutes'
+        and span_tags -> 'pwlen' = '25'
+
+```
+
+Here we are limiting our scope to spans within last 30 minutes that have a tag `pwlen` with exact value `25`. We can specify as many of these as we want combining them with logical operations:
+
+```SQL
+select *
+    from span
+    where 
+            start_time >= now() - interval '30 minutes'
+        and span_tags -> 'pwlen' = '25'
+        and resource_tags -> 'telemetry.sdk.name' = '"opentelemetry"'
+```
+
+Notice the single **and** double-quoted string `'"opentelemtry"'`. This is an artefact of PostgreSQL's `->` operator returning a `jsonb` and thus expecting a `jsonb` on the right side of the `=`. This can be worked around by using the more appropriate `->>` operator that returns text, but unfortunately that operator is not supported in the current implementation of `tag_map` type. It is planned for Promscale 1.0.0, so stay tuned!
+
+
+## 3.3. Joins <a name="para-3-3"></a>
+
+One of the main reasons to use SQL database as a backbone of your observability stack is to be able to conduct correlational analysis of different types of data, namely metrics and spans. And Promscale enables you to do exactly that.
+
+Suppose we want to see how error rate of our service correlates with the overall memory consumption of a container. It's as easy as joining the corresponding metric table with the aforementioned `span` view:
+
+```SQL
+select
```

Comment:
Have you tested the query to make sure it works?

I think what would be more interesting would be to list containers and memory consumption for those containers that have a high number of errors. Then you can inspect the values and see if there is indeed a correlation.

Reply:
I didn't, no. It'd be great to have a place where one could test these, but the demo instance that was available during 0.5.0 was unavailable for me ever since the release.

Reply:
We decided to remove this query for now